### PR TITLE
chore: add more precommit hooks

### PR DESCRIPTION
**Pull request by fpgmaas**
**PR Checklist**

- [x] A description of the changes is added to the description of this PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added or modified a feature, documentation in `docs` is updated

**Description of changes**

Add more precommit hooks

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -34,4 +34,4 @@ Steps to reproduce the behavior:
 
 <!-- Check the box below with an `x` if you are planning to create a PR for this. -->
 
-- [ ] I'm planning to make a pull-request
\ No newline at end of file
+- [ ] I'm planning to make a pull-request
```
Comment (SemyonSinchenko) [inline .github/ISSUE_TEMPLATE/bug_report.md:37]: What were changed in this line?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,4 +1,15 @@
...
+
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: "v4.6.0"
+    hooks:
+      - id: check-case-conflict
+      - id: check-merge-conflict
+      - id: check-toml
+      - id: check-yaml
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
```
Comment (SemyonSinchenko) [inline .pre-commit-config.yaml:11]: There is [something in docs](https://github.com/pre-commit/pre-commit-hooks/tree/main?tab=readme-ov-file#trailing-whitespace) about a potential problem with markdown. Should we add an option they mentined in docs?

Comment (fpgmaas) [reply .pre-commit-config.yaml:11]: Good idea, done!

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 1 above)
Comment (fpgmaas) [reply .github/ISSUE_TEMPLATE/bug_report.md:37]: to be honest; I don't see it either 😕

Comment (SemyonSinchenko) [reply .github/ISSUE_TEMPLATE/bug_report.md:37]: Let's hope we won't face strange diffs in the future with these plugins.

**Review by SemyonSinchenko — APPROVED** [review]

**Comment by SemyonSinchenko** [comment]
@fpgmaas feel free to merge! Nice work!

**Pull request merged**